### PR TITLE
fix: make uploadFile throw for non-existent files

### PR DESCRIPTION
**Pull request by paullewis**

### CODE REVIEW

File: src/JSHandle.ts [hunk 1]
```diff
@@ -291,7 +291,24 @@ export class ElementHandle extends JSHandle {
...
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const {promisify} = require('util');
+    const access = promisify(fs.access);
+
+    // Locate all files and confirm that they exist.
+    const files = await Promise.all(filePaths.map(async filePath => {
+      const resolvedPath: string = path.resolve(filePath);
+      try {
+        await access(resolvedPath);
+      } catch (e) {
```
Comment (mathiasbynens) [inline src/JSHandle.ts]: nit: e -> error

File: src/JSHandle.ts [hunk 2]
```diff
@@ -291,7 +291,24 @@ export class ElementHandle extends JSHandle {
     // the cost unnecessarily.
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const path = require('path');
-    const files = filePaths.map(filePath => path.resolve(filePath));
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const fs = require('fs');
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const {promisify} = require('util');
+    const access = promisify(fs.access);
```
Comment (mathiasbynens) [inline src/JSHandle.ts:298]: Should we use `fs.constants.R_OK` to ensure the file doesn't just exist but also that it is readable? WDYT?

**Review by mathiasbynens — CHANGES_REQUESTED** [review]
LGTM % question + nit

File: src/JSHandle.ts (hunk 1 above)
Comment (paullewis) [reply src/JSHandle.ts]: Done

File: src/JSHandle.ts (hunk 2 above)
Comment (paullewis) [reply src/JSHandle.ts:298]: Done

**Pull request merged**